src/test/java/com/ite/itea/ecommerce/domain/retail/CustomProductTest.java

```diff
@@ -0,0 +1,46 @@
+package com.ite.itea.ecommerce.domain.retail;
+
+import com.ite.itea.ecommerce.domain.core.EuroPrice;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class CustomProductTest {
+
+    @Test
+    void shouldCalculatePrice() {
+        final var product = buildCustomProduct();
+        product.setProductParts(List.of(
+                buildProductPart(EuroPrice.ofCents(100), 1),
+                buildProductPart(EuroPrice.ofCents(200), 2),
+                buildProductPart(EuroPrice.ofCents(500), 3)
+        ));
+
+        assertThat(product.price()).isEqualTo(EuroPrice.ofCents(2000));
+    }
+
+    @Test
+    void shouldCalculateZeroPriceWhenNoProductPartsExisting() {
+        final var product = buildCustomProduct();
+        assertThat(product.price()).isEqualTo(EuroPrice.zero());
+    }
+
+
+    @Test
+    void shouldCalculateZeroPriceWhenProductPartsEmpty() {
```

Comment:
I think we are testing a technical detail here (null vs. empty) that has no meaning in the domain. I would instead make sure that the getter returns an empty list even if null may be allowed internally.